addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -105,30 +105,28 @@ var SnippetEditor = Widget.extend({
             $customize.find('.oe_snippet_remove').addClass('d-none');
         }
 
-        var _animationsCount = 0;
-        var postAnimationCover = _.throttle(() => this.cover(), 100);
-        this.$target.on('transitionstart.snippet_editor, animationstart.snippet_editor', () => {
-            // We cannot rely on the fact each transition/animation start will
-            // trigger a transition/animation end as the element may be removed
-            // from the DOM before or it could simply be an infinite animation.
-            //
-            // By simplicity, for each start, we add a delayed operation that
-            // will decrease the animation counter after a fixed duration and
-            // do the post animation cover if none is registered anymore.
-            _animationsCount++;
-            setTimeout(() => {
-                if (!--_animationsCount) {
-                    postAnimationCover();
-                }
-            }, 500); // This delay have to be huge enough to take care of long
-                     // animations which will not trigger an animation end event
-                     // but if it is too small for some, this is the job of the
-                     // animation creator to manually ask for a re-cover
-        });
-        // On top of what is explained above, do the post animation cover for
-        // each detected transition/animation end so that the user does not see
-        // a flickering when not needed.
-        this.$target.on('transitionend.snippet_editor, animationend.snippet_editor', postAnimationCover);
+        this._animationsCount = 0;
+        this._animationTimeoutHandler = null;
+        this._postAnimationHandler = null;
+        this.$target.on(
+            'transitionstart.snippet_editor, animationstart.snippet_editor',
+            () => this._manageAnimationAndOverlay(1)
+        );
+        this.$target.on(
+            'transitionend.snippet_editor, animationend.snippet_editor',
+            () => this._manageAnimationAndOverlay(-1)
+        );
+        // Added cancel events as separate events as not yet well supported on every browser
+        // but worth adding for those where it is supported. It will avoid having the overlay
+        // reappearing when changing rapidly the animation.
+        this.$target.on(
+            'transitioncancel.snippet_editor',
+            () => this._manageAnimationAndOverlay(-1)
+        );
+        this.$target.on(
+            'animationcancel.snippet_editor',
+            () => this._manageAnimationAndOverlay(-1)
+        );
```

Comment:
Can't you add those on the same line as transitionend and animationend events? Or at least regroup those two if you want the comment to be aside?

Reply:
Previously I added them to the transitionend and animationend. But it seems that then if the browser don't handle them, the whole expression is ignored (tested on chrome). This is why I put them separately. I can group them but again the support is not the same (animationcancel is less supported than transistioncancel).

Reply:
Strange, but ok